src/store/modules/orderLookup/actions.ts

```diff
@@ -0,0 +1,357 @@
+          order.orderDate = order.doclist.docs[0].orderDate
+          order.orderStatusId = order.doclist.docs[0].orderStatusId
+          order.orderStatusDesc = order.doclist.docs[0].orderStatusDesc
+
+          shipmentMethodTypeIds.push(order.doclist.docs[0].shipmentMethodTypeId)
+          return order
+        })
+
+        orderCount = resp.data.grouped.orderId.ngroups;
+        itemCount = resp.data.grouped.orderId.matches;
+
+        const status = new Set();
+        const orderItems = [] as any;
+
+        orders.map((order: any) => {
+          status.add(order.orderStatusId)
+          order.doclist.docs.map((item: any) => {
+            status.add(item.orderItemStatusId)
+            orderItems.push(item)
+          })
+        })
+
+        // Added check as we only want to fetch the facets when we making first request call and do not need to fetch facets on infinite scroll
+        if(!params?.viewIndex || params.viewIndex == 0) {
+          const facilities = resp.data.facets?.facilityNameFacet?.buckets.map((bucket: any) => bucket.val)
+          const productStores = resp.data.facets?.productStoreIdFacet?.buckets.map((bucket: any) => bucket.val)
+          const channels = resp.data.facets?.salesChannelDescFacet?.buckets.map((bucket: any) => bucket.val)
+          const statuses = resp.data.facets?.orderStatusDescFacet?.buckets.map((bucket: any) => bucket.val)
+
+          commit(types.ORDERLOOKUP_CHANNEL_OPTIONS_UPDATED, channels);
+          commit(types.ORDERLOOKUP_PRODUCT_STORE_OPTIONS_UPDATED, productStores);
+          commit(types.ORDERLOOKUP_FACILITY_OPTIONS_UPDATED, facilities);
+          commit(types.ORDERLOOKUP_STATUS_OPTIONS_UPDATED, statuses);
+        }
+
+        if (query.json.params.start && query.json.params.start > 0) stateOrders = stateOrders.concat(orders)
+        else stateOrders = orders
+        this.dispatch("product/getProductInformation", { orders });
+        this.dispatch("util/fetchShipmentMethodTypeDesc", shipmentMethodTypeIds)
+      } else {
+        showToast(translate("Failed to fetch orders"));
+      }
+    } catch(error) {
+      logger.error(error)
+      showToast(translate("Failed to fetch orders"));
+    }
+    commit(types.ORDERLOOKUP_LIST_UPDATED, { orders: stateOrders, orderCount, itemCount });
+    return resp;
+  },
+
+  async getOrderDetails({ commit }, orderId) {
+    let order = {} as any;
+
+    try {
+      const orderHeader = await OrderLookupService.performFind({
+        inputFields: {
+          orderId
+        },
+        viewSize: 1,
+        entityName: "OrderHeader"
+      })
+
+      if(hasError(orderHeader) && orderHeader.data.count <= 0) {
+        throw "Failed to fetch order information"
+      }
+
+      const apiPayload = [{
+        inputFields: {
+          orderId,
+          roleTypeId: "BILL_TO_CUSTOMER"
+        },
+        viewSize: 1,
+        filterByDate: "Y",
+        fieldList: ["partyId", "roleTypeId"],
+        entityName: "OrderRole"
```

Comment:
We can use OrderHeaderAndRoles view entity, to get the combined detail from OrderHeader and OrderRole.

Reply:
Or I think we should use getOrderDetails API. Currently we can get orderHeader, items, customer, payment preferences from it. So at least we can replace three calls to fetch orderHeader, OrderRole and OrderPaymentPreference with one call to getOrderDetail API. Later on we can enhance the service to return other order detail as well.